HW5/Main.java

```diff
@@ -0,0 +1,39 @@
+package HW5;
+
+public class Main {
+
+    public static void main(String[] args) {
+
+        Greenhouse greenhouse = new Greenhouse() {
+            @Override
+            void changeTemperature(int temperature) {
+                System.out.println("Temperature system is broken! Can not change temperature at the moment.");
+            }
+        };
+
+        Bonsai bonsai = new Bonsai(1);
+        bonsai.updateNationRegion("China");
+
+        Raspberry raspberry = new Raspberry(2);
+        raspberry.updateNationRegion("Canada");
+        raspberry.harvest();
+
+        Rose rose = new Rose(3);
+        rose.thorns = new Rose.Thorns();
+        rose.thorns.addThorn();
```

Comment:
if I understood correctly, you want to use Thorns as example of nested class. If yes it should look like Rose.Thorns.addThorn();. If you want it to be inner it should not be static

Reply:
I wanted to use Thorns as a nested class. Updated to Rose.Thorns.addThorn();